hypothesis-python/src/hypothesis/strategies/_internal/numbers.py

```diff
@@ -182,97 +180,84 @@ def integers(
+        max_value: float = math.inf,
+        allow_nan: bool = True,
+    ):
         super().__init__()
-        assert isinstance(allow_infinity, bool)
         assert isinstance(allow_nan, bool)
-        assert isinstance(allow_subnormal, bool)
-        assert width in (16, 32, 64)
-        self.allow_infinity = allow_infinity
+        self.min_value = min_value
+        self.max_value = max_value
         self.allow_nan = allow_nan
-        self.allow_subnormal = allow_subnormal
-        self.width = width
 
-        self.nasty_floats = [
-            float_of(f, self.width) for f in NASTY_FLOATS if self.permitted(f)
-        ]
+        self.nasty_floats = [f for f in NASTY_FLOATS if self.permitted(f)]
         weights = [0.2 * len(self.nasty_floats)] + [0.8] * len(self.nasty_floats)
-        self.sampler = d.Sampler(weights)
+        self.sampler = d.Sampler(weights) if self.nasty_floats else None
+
+        self.forced_sign_bit: Optional[int] = None
+        if _sign_aware_lte(0.0, max_value):
+            self.pos_clamper = make_float_clamper(max(0.0, min_value), max_value)
+            if _sign_aware_lte(0.0, min_value):
+                self.forced_sign_bit = 0
+        if not _sign_aware_lte(0.0, min_value):
+            self.neg_clamper = make_float_clamper(-min(-0.0, max_value), -min_value)
+            if not _sign_aware_lte(0.0, max_value):
+                self.forced_sign_bit = 1
 
     def __repr__(self):
-        return "{}(allow_infinity={}, allow_nan={}, width={})".format(
-            self.__class__.__name__, self.allow_infinity, self.allow_nan, self.width
+        return "{}(min_value={}, max_value={}, allow_nan={})".format(
+            self.__class__.__name__,
+            self.min_value,
+            self.max_value,
+            self.allow_nan,
         )
 
     def permitted(self, f):
         assert isinstance(f, float)
-        if not self.allow_infinity and math.isinf(f):
-            return False
-        if not self.allow_nan and math.isnan(f):
-            return False
-        if self.width < 64:
-            try:
-                float_of(f, self.width)
-            except OverflowError:
-                return False
-        if not self.allow_subnormal and 0 < abs(f) < width_smallest_normals[self.width]:
-            return False
-        return True
+        if math.isnan(f):
+            return self.allow_nan
+        return _sign_aware_lte(self.min_value, f) and _sign_aware_lte(f, self.max_value)
 
     def do_draw(self, data):
         while True:
             data.start_example(FLOAT_STRATEGY_DO_DRAW_LABEL)
-            i = self.sampler.sample(data)
+            i = self.sampler.sample(data) if self.sampler else 0
             if i == 0:
-                result = flt.draw_float(data)
+                is_negative = data.draw_bits(1, forced=self.forced_sign_bit)
+                data.start_example(flt.DRAW_FLOAT_LABEL)
+                result = flt.lex_to_float(data.draw_bits(64))
```

Comment:
> * For consistency with the database repr of prior-to-this-PR unbounded floats, let's order that as "float contents then sign bit".  I think this just consists of undoing the changes to `draw_float()` and `write_float()`, then directly messing with the sign here (`if clamped != result:`) if it wasn't what we'd force

Ah! Let me ask about this point right away, as it probably impacts the most number of places. I moved the sign bit forward, because we effectively lost a prefix byte that was coming from `OneOfStrategy` -- previously, we were unioning two float strategies, one for each sign. Sadly, though, that was only for bounded floats, the unbounded case didn't have that byte. So I'm **still** not compatible with old examples. That said, when I did the naive thing of leaving the sign at the end, lots of tests started failing; I presumed that the early sign decision was important to the way the search or shrinking happens, but I didn't really make an attempt to confirm.

Anyway, I could imagine lots of paths forward from here, but want your fully informed opinion before I dig in!

Reply:
Cool; okay, I think I've made the right changes for the example bounds! But don't take my word for it :)